File: frontend/types/event.ts
export interface Event {
  id: number;
  title: string;
  description: string;
  date: string; // ISO 8601
  max_capacity: number;
  registered_count: number;
  available_spots: number;
  created_at: string;
  updated_at: string;
}

export interface EventCreate {
  title: string;
  description: string;
  date: string;
  max_capacity: number;
}

export interface EventUpdate {
  title?: string;
  description?: string;
  date?: string;
  max_capacity?: number;
}

File: backend/app/models/event.py
from dataclasses import dataclass, field
from datetime import datetime, timezone


@dataclass
class Event:
    """Represents an event in the system."""

    id: int
    title: str
    description: str
    date: datetime
    max_capacity: int
    created_at: datetime = field(default_factory=lambda: datetime.now(timezone.utc))
    updated_at: datetime = field(default_factory=lambda: datetime.now(timezone.utc))